### ISSUE TITLE
Lens flare effect green flash, corrupt FMVs

### ISSUE THREAD

**Issue by 71knight** [hrydgard/ppsspp#16075]
### Game or games this happens in

Burnout dominator, need for speed shift, maybe more

### What area of the game

During full motion videos before racing, also lens flare has green boxes in it briefly.

### What happens

Full motion videos have digital artifact errors/ random digital error squares, sometimes almost covering complete screen.  Lens flare has momentary green squares in it.  Best seen when viewing car model at car selection screen and rotating car.  I believe the developer said this was caused by not implementing Z-upload.

### What should happen

Videos should play back perfect.  Lens flare effect shouldn't have green boxes in it briefly.

### Platform

Android 11

### Mobile phone model or graphics card

OnePlus 7t with adreno 640.

### PPSSPP version affected

1.13.2 1091

### Last working version

Dunno

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [ ] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
OK, not only was checking the CLUT usage flag of the framebuffer not reliable (of course, one frame late), but we end up resizing the clut framebuffer immediately after creating for some reason, which loses the just-uploaded z buffer. Fixing that.

**Comment by hrydgard** [comment]
Oops, looks like I broke this again in #16072 . Many moving parts.. Should be an easy fix.

**Comment by 71knight** [comment]
Sorry for bad news but sunspot is visible through mountains and tunnels.

**Comment by hrydgard** [comment]
Not for me on PC. But I'll take a look on Android. Thanks

**Comment by Gamemulatorer** [comment]
> Sorry for bad news but sunspot is visible through mountains and tunnels.

What backend?

**Comment by 71knight** [comment]
Nevermind....the sunspot is fixed along with the lens flare in the latest build.  The full motion video is still corrupt in burnout dominator and need for speed shift.  The problem appears to be because of live rendered graphics (X and O button, options, etc.) over the full motion video and the buffers aren't playing nice with each other I guess.  I played a full motion video with no graphic overlays and it played perfectly.  I even tried another ISO of the same game and it still was broken.  Maybe the lens flare and sun fix had something to do with it maybe?  Great job on fixing the lens flare and sun occlusion.

**Comment by hrydgard** [comment]
Thanks. I don't understand which FMV is looking corrupt though, can you make a screenshot, or tell me which one and how to get it to play?

**Comment by 71knight** [comment]
On need for speed shift you can see it when you're selecting different options in the menu before a race. 
 You should see the background glitch when selecting options at the beginning of the game.

**Comment by hrydgard** [comment]
Alright, that kind of glitch is, while definitely a bug, definitely unrelated to the green lens flare and similar.

NFS Shift does seem to have other issues though. The bloom is horribly ugly although that's an artifact of resoultion, and gets better with "Lower resolution for effects". However, billboards in the game seem to be textured with noise or grass or something, that can't be right.

**Comment by 71knight** [comment]
I've just watched this game being played on a real PSP on YouTube.  From what I can tell all the graphics are correct!  I don't notice any problem at all with the billboards they look perfect.  The only problems are the aforementioned fmvs backgrounds and I've noticed the nitro effect is a little heavy.... It has a slight stutter at the beginning and at the end and I have all my CPU cores maxed out on my phone (SD 855+)  I don't know if that can be optimized or not.
